src/main/java/org/apache/ibatis/migration/commands/NewCommand.java

```diff
@@ -35,35 +44,87 @@ public void execute(String... params) {
     if (paramsEmpty(params)) {
       throw new MigrationException("No description specified for new migration.");
     }
-    String description = params[0];
-    Properties variables = new Properties();
-    variables.setProperty("description", description);
-    existingEnvironmentFile();
-    String filename = getNextIDAsString() + "_" + description.replace(' ', '_') + ".sql";
 
-    if (options.getTemplate() != null) {
-      copyExternalResourceTo(options.getTemplate(), Util.file(paths.getScriptPath(), filename), variables);
-    } else {
+
+    Hook hook = createNewMigrationHook();
+
+    Properties variables = getVariables();
+    variables.setProperty("description", params[0]);
+
+
+    Map<String, Object> hookBindings = createBinding(params);
+    {
+
+      Reader templateReader = getTemplateReader();
+      Change change = (Change) hookBindings.get("change");
+      hook.before(hookBindings);
+      File changeFile = new File(change.getFilename());
       try {
-        String customConfiguredTemplate = getPropertyOption(CUSTOM_NEW_COMMAND_TEMPLATE_PROPERTY);
-        if (customConfiguredTemplate != null) {
-          copyExternalResourceTo(migrationsHome() + "/" + customConfiguredTemplate,
-              Util.file(paths.getScriptPath(), filename), variables);
-        } else {
-          copyDefaultTemplate(variables, filename);
-        }
-      } catch (FileNotFoundException e) {
-        printStream
-            .append("Your migrations configuration did not find your custom template.  Using the default template.");
-        copyDefaultTemplate(variables, filename);
+        copyTemplate(templateReader, changeFile, variables);
+      } catch (IOException e) {
+        throw new MigrationException("Unable to create template file " + changeFile.getAbsolutePath());
       }
+
+      hook.after(hookBindings);
     }
     printStream.println("Done!");
     printStream.println();
+
+  }
+
+  private Reader getTemplateReader() {
```

Comment:
This whole function is meant to normalize finding the reader for the template so that the calling code can seperate the `what` from the `how` `what: I need to get a template. How: by checking a whole lot of config`